sinks/signalfx/signalfx.go

```diff
@@ -183,11 +201,91 @@ func NewClient(endpoint, apiKey string, client *http.Client) DPClient {
 	return httpSink
 }
 
+func MigrateConfig(conf *veneur.Config) error {
+	if conf.SignalfxAPIKey.Value != "" {
+		return nil
+	}
+	if conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod == "" {
+		conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod = "10m"
+	}
+	dynamicPerTagAPIKeysRefreshPeriod, err :=
+		time.ParseDuration(conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod)
+	if err != nil {
+		return err
+	}
+	conf.MetricSinks = append(conf.MetricSinks, struct {
+		Kind   string      "yaml:\"kind\""
+		Name   string      "yaml:\"name\""
+		Config interface{} "yaml:\"config\""
+	}{
+		Kind: "signalfx",
+		Name: "signalfx",
+		Config: SignalFxSinkConfig{
+			APIKey:                            conf.SignalfxAPIKey,
+			DynamicPerTagAPIKeysEnable:        conf.SignalfxDynamicPerTagAPIKeysEnable,
+			DynamicPerTagAPIKeysRefreshPeriod: dynamicPerTagAPIKeysRefreshPeriod,
+			EndpointAPI:                       conf.SignalfxEndpointAPI,
+			EndpointBase:                      conf.SignalfxEndpointBase,
+			FlushMaxPerBody:                   conf.SignalfxFlushMaxPerBody,
+			HostnameTag:                       conf.SignalfxHostnameTag,
+			MetricNamePrefixDrops:             conf.SignalfxMetricNamePrefixDrops,
+			MetricTagPrefixDrops:              conf.SignalfxMetricTagPrefixDrops,
+			PerTagAPIKeys:                     conf.SignalfxPerTagAPIKeys,
+			VaryKeyBy:                         conf.SignalfxVaryKeyBy,
+		},
+	})
+	return nil
+}
+
+func CreateSignalFxSink(
+	server *veneur.Server, name string, config veneur.Config,
+	sinkConfig interface{},
+) (sinks.MetricSink, error) {
+	signalFxConfig := SignalFxSinkConfig{}
+	err := util.DecodeConfig(sinkConfig, &signalFxConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	log := veneur.GetLogger()
```

Comment:
Some things we pass from `createMetricSinks` as an argument, some things we export a function from the `veneur` package. Is it useful to keep consistency on one of the two options? On the flip is it useful to explicitly not keep consistency?

Reply:
Actually, I really like how adding the logger to the signature turned out.

Reply:
Yeah, I had `.WithField("sink", sinkName)` in mind when I wrote this comment :)